### PR TITLE
WIP: Deprecate `Post.section_link()`.

### PR DESCRIPTION
**Pull request by felixfontein**
WIP for #2086.

Where should this go in `CHANGES.txt`? Is this a feature or a bug? :)

### CODE REVIEW

File: nikola/post.py [hunk 1]
```diff
@@ -744,7 +744,7 @@ def section_color(self, lang=None):
         return utils.colorize_str_from_base_color(slug, base)
 
     def section_link(self, lang=None):
-        """Return the link to the post's section."""
+        """Return the link to the post's section (deprecated)."""
         if lang is None:
```
Comment (Kwpolska) [inline nikola/post.py]: could use a deprecation warning (`logger.warning` if we can access that)

**Comment by Kwpolska** [comment]
Add a different changelog section for this, at the end, named *Deprecations*, I guess?

Also, please make a commit with `fix #2086` in the message.

File: nikola/post.py (hunk 1 above)
Comment (felixfontein) [inline nikola/post.py]: `utils.LOGGER` is available.

**Comment by Kwpolska** [comment]
Thanks!

**Comment by felixfontein** [comment]
You're welcome.

**Pull request merged**